broadinstitute/pearl
# [JN-510] persisting answer mappings along with surveys

This fixes a bug where answer mappings (the things that map some survey questions to profile fields) weren't being copied forward to new versions of the survey.  This is because the API service expects that the new version will specify any answer mappings, but the answer mappings for the existing survey were never being sent to the frontend.  To solve this, as well as to move our admin tool into more modular loading patterns, the survey editing component now loads the survey itself, along with the answer mappings, rather than relying on the initial portal blob fetch.    

This also does a drive-by fix of the "cancel" button navigation on the survey editing page

TO TEST:
1. repopulate ourhealth (just to make sure you don't have stale survey copies floating around)
2. in your DB, run `select * from answer_mapping;` and observe there are 9 rows.
3. edit the 'basics' survey `https://localhost:3000/ourhealth/studies/ourheart/env/sandbox/forms/surveys/oh_oh_basicInfo?readOnly=false`
4. save it
5. confirm the version is updated
6. in your DB, run `select * from answer_mapping;` and confirm there are now 18 rows.
